src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -2058,13 +2061,19 @@ public int hashCode() {
             if (firstOnly) return ret;
         }
 
-        if (this.getAlignmentStart() != NO_ALIGNMENT_START && this.getIndexingBin() != null &&
-                this.computeIndexingBin() != this.getIndexingBin()) {
-            if (ret == null) ret = new ArrayList<>();
-            ret.add(new SAMValidationError(SAMValidationError.Type.INVALID_INDEXING_BIN,
-                    "bin field of BAM record does not equal value computed based on alignment start and end, and length of sequence to which read is aligned",
-                    getReadName()));
-            if (firstOnly) return ret;
+        if (this.getAlignmentStart() != NO_ALIGNMENT_START) {
+            final SAMSequenceRecord sequence = getHeader() != null ? getHeader().getSequence(getReferenceName()) : null;
+            if (sequence != null && isReferenceTooLargeForBAI(sequence)) {
+                if (getValidationStringency() != ValidationStringency.SILENT) {
+                    LOG.warn(String.format("Reference length is too large for BAM bin field for %s record. Values in the bin field could be incorrect.", getReadName()));
```

Comment:
"Values in the bin field could be incorrect" is pretty vague. If we're declaring that the right thing to do for this case is to use a narrowed value (the one the BAMRecordCodec now emits), then it seems like we should check for that value here, and use a different log message for the case where the value is the "expected" truncated value, vs. a completely incorrect value. Also I think it would be useful to include the reference contig name, and possibly even the bin values in the case where its completely incorrect.

Reply:
We should have emphasize that in case reference length is larger than BAI index can handle, bin values are always incorrect. So there are no different cases.
We have clarified warning message, now it shows that all bin field values are incorrect.